### PR TITLE
Subaru: cleanup tests

### PR DESCRIPTION
**Pull request by jnewb1**

### CODE REVIEW

**Comment by jnewb1** [comment]
On using enums: this is how ford port does it. Enum would be fine but we probably want to be consistent. What do you prefer?

https://github.com/commaai/panda/blob/6d2b70a1475922983f59c57e68e68a6f61099268/tests/safety/test_ford.py#L11-L21

File: tests/safety/test_hyundai_canfd.py [hunk 1]
```diff
@@ -95,7 +96,8 @@ class TestHyundaiCanfdHDA1Base(TestHyundaiCanfdBase):
 
   @classmethod
   def setUpClass(cls):
-    if cls.__name__ in ("TestHyundaiCanfdHDA1", "TestHyundaiCanfdHDA1AltButtons") or cls.__name__.endswith('Base'):
+    super().setUpClass()
```
Comment (jnewb1) [inline tests/safety/test_hyundai_canfd.py:99]: Called base setup functions here, which removes need for the "Base" check

**Comment by jnewb1** [comment]
> On using enums: this is how ford port does it. Enum would be fine but we probably want to be consistent. What do you prefer?
> 
> https://github.com/commaai/panda/blob/6d2b70a1475922983f59c57e68e68a6f61099268/tests/safety/test_ford.py#L11-L21

Maybe it would be good to refactor one port fully to exactly how we want future ports to be, and then try and use that as a standard?

File: tests/safety/common.py [hunk 2]
```diff
@@ -190,6 +190,7 @@ class TorqueSteeringSafetyTestBase(PandaSafetyTestBase, abc.ABC):
 
   @classmethod
   def setUpClass(cls):
+    super().setUpClass()
```
Comment (sshane) [inline tests/safety/common.py:193]: don't think this does anything. the endswith check is in `PandaSafetyTest`, not the base class. forgot why I didn't just put endswith base in the `PandaSafetyTestBase` class, can you see if there's any issues? can be separate from this subaru-specific PR

File: tests/safety/common.py [hunk 3]
```diff
@@ -335,6 +335,7 @@ class DriverTorqueSteeringSafetyTest(TorqueSteeringSafetyTestBase, abc.ABC):
...
```
Comment (sshane) [inline tests/safety/common.py:338]: this doesn't seem to do anything at the moment

**Pull request merged**